src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,55 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.ExpectedException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_CHARACTERS = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkWrongValue(user);
+        return storageDao.add(user);
+    }
+
+    private void checkWrongValue(User user) {
+
+        if (user == null) {
+            throw new ExpectedException("User can't be null");
+        }
```

Comment:
I think you can combine null and value validation (cause if Login is null it means it less than 6 character) same with password and age